azkaban-common/src/main/java/azkaban/metric/IMetricEmitter.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.metric;
+
+/**
+ * Interface for metric emitters
+ */
+public interface IMetricEmitter {
+  void reportMetric(final IMetric<?> metric) throws Exception;
```

Comment:
Does it make sense to introduce MetricException?  When a method throws Exception, it doesn't tell the caller much about the type of exception it can throw.